### DISCUSSION TITLE
What should I use for media_prior argument in mmm.fit function?

### DISCUSSION THREAD

**Discussion by AruunnAnanda** [google/lightweight_mmm#117] [Q&A]
Hi there,

I understand that for the media prior argument I need to pass one value for each media spend channel in the form of an array. 

Now my question is what should that value be?

Currently, I used a randomly typed number for all the channels as you can see below and executed the code. Is this wrong? What is your recommendation?

Thanks!

![image](https://user-images.githubusercontent.com/107944288/206422436-7e6b447d-c3c1-4e71-9d44-3e3c45102152.png)

![image](https://user-images.githubusercontent.com/107944288/206422469-12c61141-6ba0-4700-a91f-69352e9dffe8.png)

**Comment by michevan** [comment]
Hi AruunnAnanda

If you have data for the total cost of the media on each channel, we would recommend using those values here for your media priors, after normalizing them with a CustomScaler as shown in the example Colabs!

**Reply by AruunnAnanda** [reply]
Okay. Would it be total costs for the entire time-frame or total costs of just training period timeframe? Thanks!

**Reply by michevan** [reply]
I think it should be just during the training period!

**Reply by danivapp** [reply]
why are total costs used? could i also pass costs as time series?

**Reply by michevan** [reply]
you have to pass total costs over the time range under consideration to generate the half-normal distributions we use for the default media channel priors. You can create custom priors though if you prefer a different distribution!

**Reply by danivapp** [reply]
How is the half-normal distribution generated form only the grand total? Is there any documentation regarding this?

**Reply by greenfrog555** [reply]
This is the [document](https://lightweight-mmm.readthedocs.io/en/latest/models.html) regarding how half-normal distribution with cost adjusted scale works. The equation is under the "media effect" section. Let us know whether it is clear.

**Reply by danivapp** [reply]
Thanks @greenfrog555, i have seen this durin my research. [Half-normal](https://en.wikipedia.org/wiki/Half-normal_distribution) is parametrized by only one parameter - its scale σ, since the location is fixed to μ=0 by the very definition of this distribution. Is the total cost passed as the scale parameter then? I seem to be missing something trivial here.

**Reply by greenfrog555** [reply]
Yes. The scaled cost for each channel will be passed as the scale in the Half-Normal distribution. The total cost across all the channels will be scaled with this function in our sample Colab. "cost_scaler = preprocessing.CustomScaler(divide_operation=jnp.mean, multiply_by=0.15)" The total cost is scaled by the mean and times a multiplier whose default is 0.15 but you can adjust it.

**Reply by aruizartg** [reply]
Hello,

I entered this discussion because I have a similar question regarding the `media_prior` parameter. Should we always calculate the total costs based on the spend of each media channel? For instance, if we're using different metrics for the `media` parameter (such as impressions, GRPs for TV, etc.), do we still determine the `media_prior` by summing up the spends, or should we aggregate the metrics used to define the `media` parameter in this scenario?

I realize this might be a straightforward question, but I want to be absolutely certain. Thank you in advance! 😄
